### PR TITLE
fixes #86 associativity of exponentiation operator

### LINKED ISSUE #86: Incorrect associativity of exponentiation operator
Bug report inspired by https://github.com/babel/babel/issues/8325

```
$ terser -V
terser 3.7.8
```
```
$ cat exponent_operator.js 
console.log((4 ** 1) ** 2, (4 ** 1) ** (1 / 2));
var one = 1;
var two = 2;
var four = 4;
console.log((four ** one) ** two, (four ** one) ** (one / two));
```
Expected:
```
$ cat exponent_operator.js | node-v10.4.1 
16 2
16 2
```
Actual:
```
$ cat exponent_operator.js | terser | node-v10.4.1 
4 4
4 4
```
```
$ cat exponent_operator.js | terser -mc | node-v10.4.1 
16 2
4 4
```

Looks like the paren issue would easily be fixed in `lib/output.js`. Good first PR for someone.

### PR DESCRIPTION
**Pull request by JamesNimlos**

### CODE REVIEW

File: test/compress/evaluate.js [hunk 1]
```diff
@@ -325,6 +325,61 @@ pow_with_number_constants: {
...
+    }
+    input: {
+        console.log((4 ** 1) ** 2, (4 ** 1) ** (1 / 2));
+    }
+    expect: {
+        console.log(16,2);
+    }
+}
+
+
```
Comment (kzc) [inline test/compress/evaluate.js]: only one line between tests please

File: test/compress/evaluate.js [hunk 2]
```diff
@@ -325,6 +325,61 @@ pow_with_number_constants: {
...
 
+pow_sequence_with_parens: {
+    input: {
+        var one = 1;
+        var two = 2;
+        var four = 4;
+        console.log((four ** one) ** two, (four ** one) ** (one / two));
+    }
+    expect: {
+        var one=1;var two=2;var four=4;console.log((four**one)**two,(four**one)**(one/two));
```
Comment (kzc) [inline test/compress/evaluate.js]: please format expect code one statement per line here - similar to input.

File: test/compress/evaluate.js [hunk 3]
```diff
@@ -325,6 +325,61 @@ pow_with_number_constants: {
...
+    }
```
Comment (kzc) [inline test/compress/evaluate.js]: please add an `expect_stdout` clause

File: test/compress/evaluate.js [hunk 4]
```diff
@@ -325,6 +325,61 @@ pow_with_number_constants: {
...
+    input: {
+        var one = 1;
+        var two = 2;
+        var four = 4;
+        console.log((four ** one) ** two, (four ** one) ** (one / two));
+    }
+    expect: {
+        console.log(16,2);
+    }
```
Comment (kzc) [inline test/compress/evaluate.js]: please add an `expect_stdout` clause

File: test/compress/evaluate.js [hunk 5]
```diff
@@ -325,6 +325,61 @@ pow_with_number_constants: {
...
+        evaluate: true,
+        toplevel: true,
+    }
+    input: {
+        console.log((4 ** 1) ** 2, (4 ** 1) ** (1 / 2));
+    }
+    expect: {
+        console.log(16,2);
+    }
+}
```
Comment (kzc) [inline test/compress/evaluate.js]: please add an `expect_stdout` clause

File: test/compress/evaluate.js [hunk 6]
```diff
@@ -325,6 +325,61 @@ pow_with_number_constants: {
...
+pow_sequence_with_parens_exact: {
+    input: {
+        console.log((4 ** 1) ** 2, (4 ** 1) ** (1 / 2));
+        var one = 1;
+        var two = 2;
+        var four = 4;
+        console.log((four ** one) ** two, (four ** one) ** (one / two));
+    }
+    expect_exact: "console.log((4**1)**2,(4**1)**(1/2));var one=1;var two=2;var four=4;console.log((four**one)**two,(four**one)**(one/two));"
+}
```
Comment (kzc) [inline test/compress/evaluate.js]: please add an `expect_stdout` clause

**Comment by kzc** [comment]
@JamesNimlos The fix looks good.

`expect_stdout` will work for each test if you also add:
```
    node_version: ">=8"
```

When you're done please squash your changes into a single commit.

**Comment by kzc** [comment]
By convention the squashed commit comment should look like this:
```
fix associativity of exponentiation operator 

fixes #86 
```

**Comment by JamesNimlos** [comment]
I've made the requested changes I think. Let me know if there are additional issues.

File: test/compress/evaluate.js [hunk 7]
```diff
@@ -325,6 +325,73 @@ pow_with_number_constants: {
...
+    options = {
+        evaluate: true,
+        toplevel: true,
+    }
+    input: {
+        console.log((4 ** 1) ** 2, (4 ** 1) ** (1 / 2));
+    }
+    expect: {
+        console.log(16,2);
+    }
```
Comment (kzc) [inline test/compress/evaluate.js]: missed an `expect_stdout` clause here

**Comment by kzc** [comment]
By convention `node_version` is the last line in each test. Other than that, it looks good. Thanks.

**Comment by kzc** [comment]
One last request: would you mind adding this additional test?

```js
pow_negative: {
    input: {
        var x = -5, y = -3, z = -2;
        console.log((-x)**(-y)**(-z), ((-x)**(-y))**(-z), (-x)**((-y)**(-z)));
    }
    expect_exact: "var x=-5,y=-3,z=-2;console.log((-x)**((-y)**-z),((-x)**-y)**-z,(-x)**((-y)**-z));"
    expect_stdout: "1953125 15625 1953125"
    node_version: ">=8"
}
```

**Comment by kzc** [comment]
Close enough. 

@fabiosantoscode Could you please commit the [`pow_negative `](https://github.com/fabiosantoscode/terser/pull/88#issuecomment-408863492) test case after merging this PR?

**Comment by kzc** [comment]
LGTM.

This PR is good to merge as is without the `pow_negative` test case.

**Comment by fabiosantoscode** [comment]
Good!

**Pull request merged**